vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/OidcSiopWallet.kt

```diff
@@ -270,139 +271,103 @@ class OidcSiopWallet(
+            runCatching { params.parameters.verifyRedirectUrl() }
+                .getOrElse { return KmmResult.failure(it) }
         }
 
-        if (params.parameters.responseType == null)
-            return KmmResult.failure<AuthenticationResponseParameters>(OAuth2Exception(Errors.INVALID_REQUEST))
-                .also { Napier.w("response_type is not specified") }
-        if (!params.parameters.responseType.contains(VP_TOKEN) && params.parameters.presentationDefinition == null)
-            return KmmResult.failure<AuthenticationResponseParameters>(OAuth2Exception(Errors.INVALID_REQUEST))
-                .also { Napier.w("vp_token not requested") }
-        val presentationDefinition =
-            params.parameters.presentationDefinition ?: params.parameters.presentationDefinitionUrl?.let {
-                remoteResourceRetriever.invoke(it)
-            }?.let { PresentationDefinition.deserialize(it).getOrNull() }
-            ?: params.parameters.scope?.split(" ")?.firstNotNullOfOrNull {
-                scopePresentationDefinitionRetriever?.invoke(it)
-            }
-        if (!params.parameters.responseType.contains(VP_TOKEN) && presentationDefinition == null)
-            return KmmResult.failure<AuthenticationResponseParameters>(OAuth2Exception(Errors.INVALID_REQUEST))
-                .also { Napier.w("vp_token not requested") }
+        val presentationDefinition = params.parameters.loadPresentationDefinition()
+        runCatching { params.parameters.verifyResponseType(presentationDefinition) }
+            .onFailure { return KmmResult.failure(it) }
         if (clientMetadata.vpFormats != null) {
-            if (clientMetadata.vpFormats.jwtVp?.algorithms != null
-                && clientMetadata.vpFormats.jwtVp?.algorithms?.contains(jwsService.algorithm.identifier) != true
-            ) return KmmResult.failure<AuthenticationResponseParameters>(OAuth2Exception(Errors.REGISTRATION_VALUE_NOT_SUPPORTED))
-                .also { Napier.w("Incompatible JWT algorithms") }
-            if (clientMetadata.vpFormats.jwtSd?.algorithms != null
-                && clientMetadata.vpFormats.jwtSd?.algorithms?.contains(jwsService.algorithm.identifier) != true
-            ) return KmmResult.failure<AuthenticationResponseParameters>(OAuth2Exception(Errors.REGISTRATION_VALUE_NOT_SUPPORTED))
-                .also { Napier.w("Incompatible JWT algorithms") }
-            if (clientMetadata.vpFormats.msoMdoc?.algorithms != null
-                && clientMetadata.vpFormats.msoMdoc?.algorithms?.contains(jwsService.algorithm.identifier) != true
-            ) return KmmResult.failure<AuthenticationResponseParameters>(OAuth2Exception(Errors.REGISTRATION_VALUE_NOT_SUPPORTED))
-                .also { Napier.w("Incompatible JWT algorithms") }
+            runCatching { clientMetadata.vpFormats.verifySupport(jwsService.algorithm) }
+                .onFailure { return KmmResult.failure(it) }
+        }
```

Comment:
This check is probably obsolete, the formats of the submitted tokens are validated at lines 435-458 of the old file version (line 319-324 of commit 88cd53f3017a6e93dfc23604f64fd095838e33e7). 

Reason: It's not important whether the relying part supports a format if no token of that format is issued

Blame: https://github.com/a-sit-plus/kmm-vc-library/commit/27da00a2402bba8d7fc51544449160fe9b361892#r142288803

Reply:
Done in https://github.com/a-sit-plus/kmm-vc-library/pull/68/commits/5cda85bcada80db90bdd352ba0092b1c3c0553a9